gui/job-details.lua

```diff
@@ -0,0 +1,425 @@
+-- Show and modify properties of jobs in a workshop or of manager orders.
+
+--[[
+    Credit goes to the author of `gui/workshop-job`.
+    This is the gui/workshop-job.lua with gui/workorder-details.lua added on top
+    and updated to the DF50.
+]]
+
+--@ module = true
+
+local overlay = require('plugins.overlay')
+
+local utils = require 'utils'
+local gui = require 'gui'
+local guimat = require 'gui.materials'
+local widgets = require 'gui.widgets'
+local dlg = require 'gui.dialogs'
+
+JobDetailsScreen = defclass(JobDetailsScreen, gui.ZScreenModal)
+
+JobDetailsScreen.ATTRS {
+    focus_path = 'job-details',
+}
+
+function JobDetailsScreen:init(args)
+    self:addviews{JobDetails(args)}
+end
+
+JobDetails = defclass(JobDetails, widgets.Window)
+
+JobDetails.ATTRS {
+    frame_title='Details',
+    resizable = true,
+    resize_min={w=50, h=20},
+    frame = { l = 10, w = 50 },
+    --
+    job = DEFAULT_NIL,
+    context = DEFAULT_NIL,
+}
+
+function JobDetails:isManagerOrder()
+    return self.context == df.job_details_context_type.MANAGER_WORK_ORDER
+end
+
+function JobDetails:init(args)
+    local status
+    if not self:isManagerOrder() then
+        status = { text = 'No worker', pen = COLOR_DARKGREY }
+        local worker = dfhack.job.getWorker(self.job)
+        if self.job.flags.suspend then
+            status = { text = 'Suspended', pen = COLOR_RED }
+        elseif worker then
+            status = { text = dfhack.TranslateName(dfhack.units.getVisibleName(worker)), pen = COLOR_GREEN }
+        end
+    end
+
+    self:addviews{
+        widgets.Label{
+            frame = { l = 0, t = 0 },
+            text = {
+                { text = df.job_type.attrs[self.job.job_type].caption }, NEWLINE, NEWLINE,
+                '  ', status
+            }
+        },
+        widgets.Label{
+            frame = { l = 0, t = 4 },
+            text = {
+                { key = 'CUSTOM_I', text = ': Input item, ',
+                  enabled = self:callback('canChangeIType'),
+                  on_activate = self:callback('onChangeIType') },
+                { key = 'CUSTOM_M', text = ': Material, ',
+                  enabled = self:callback('canChangeMat'),
+                  on_activate = self:callback('onChangeMat') },
+                { key = 'CUSTOM_T', text = ': Traits',
+                  enabled = self:callback('canChangeTrait'),
+                  on_activate = self:callback('onChangeTrait') }
+            }
+        },
```

Comment:
these aren't clickable. Try making them individual `HotkeyLabel`s

Reply:
Did that and I think discovered a bug: it highlights every hotkeylabel up to mouse X position (but correct action is executed)

![image](https://github.com/DFHack/scripts/assets/348150/62f30861-51a7-4ec7-aa75-50cd04dc62dc)


![image](https://github.com/DFHack/scripts/assets/348150/f92d21de-6eab-44da-bf43-359b894251cd)

Reply:
That's because the label frame is larger than the label text. Try setting `auto_width=true` or setting an explicit width